polkadot/zombienet-sdk-tests/tests/smoke/coretime_revenue.rs

```diff
@@ -0,0 +1,514 @@
+			&bob,
+		)
+		.await?
+		.wait_for_finalized_success()
+		.await?;
+
+	let order = r
+		.find_first::<rococo_api::on_demand_assignment_provider::events::OnDemandOrderPlaced>()?
+		.unwrap();
+
+	// As there's no spot traffic, Bob will only pay base fee
+
+	assert_eq!(order.spot_price, ON_DEMAND_BASE_FEE);
+
+	// Somewhere below this point, revenue is generated and is teleported to the PC (that happens
+	// once a timeslice so we're not ready to assert it yet, let's just account). That checks out
+	// tokens from the RC and mints them on the PC.
+
+	total_issuance.1 += ON_DEMAND_BASE_FEE;
+
+	// As soon as the PC receives the tokens, it divides them half by half into system and private
+	// contributions (we have 3 cores, one is leased to Coretime itself, one is pooled by the
+	// system, and one is pooled by Alice).
+
+	// Now we're waiting for the moment when Alice may claim her revenue
+
+	log::info!("Waiting for Alice's revenue to be ready to claim");
+
+	let claims_ready = wait_for_para_event(
+		para_events.clone(),
+		"Broker",
+		"ClaimsReady",
+		|e: &broker_events::ClaimsReady| e.when == pooled.region_id.begin,
+	)
+	.await;
+
+	// The revenue should be half of the spot price, which is equal to the base fee.
+
+	assert_eq!(claims_ready.private_payout, ON_DEMAND_BASE_FEE / 2);
+
+	// By this moment, we're sure that revenue was received by the PC and can assert the total
+	// issuance
+
+	assert_total_issuance(relay_client.clone(), para_client.clone(), total_issuance).await;
+
+	// Alice claims her revenue
+
+	log::info!("Alice is going to claim her revenue");
+
+	para_client
+		.tx()
+		.sign_and_submit_default(
+			&coretime_api::tx().broker().claim_revenue(pooled.region_id, pooled.duration),
+			&alice,
+		)
+		.await?;
+
+	let claim_paid = wait_for_para_event(
+		para_events.clone(),
+		"Broker",
+		"RevenueClaimPaid",
+		|e: &broker_events::RevenueClaimPaid| e.who == alice_acc,
+	)
+	.await;
+
+	log::info!("Revenue claimed, waiting for 2 timeslices until the system revenue is burnt");
+
+	assert_eq!(claim_paid.amount, ON_DEMAND_BASE_FEE / 2);
+
+	// As for the system revenue, it is teleported back to the RC and burnt there. Those burns are
+	// batched and are processed once a timeslice, after a new one starts. So we have to wait for
+	// two timeslice boundaries to pass to be sure the teleport has already happened somewhere in
+	// between.
+
+	let _: coretime_api::broker::events::SaleInitialized =
+		wait_for_para_event(para_events.clone(), "Broker", "SaleInitialized", |_| true).await;
+
+	total_issuance.0 -= ON_DEMAND_BASE_FEE / 2;
+	total_issuance.1 -= ON_DEMAND_BASE_FEE / 2;
+
+	let _: coretime_api::broker::events::SaleInitialized =
+		wait_for_para_event(para_events.clone(), "Broker", "SaleInitialized", |_| true).await;
+
+	assert_total_issuance(relay_client.clone(), para_client.clone(), total_issuance).await;
+
+	log::info!("Test finished successfuly");
+
+	Ok(())
+}
```

Comment:
Nice! Are more complicated scenarios, e.g. interlaced cores/split regions already sufficiently covered with unit tests? If not, would be good to make the test scenario here a bit more involved.

Reply:
They are covered by unit tests. It would still be good to add them to this test as well, but it already takes ages to finish, so I left just common logic here.